src/pal/src/misc/stackstring.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+#include "pal/malloc.hpp"
+#include "pal/dbgmsg.h"
+
+SET_DEFAULT_DEBUG_CHANNEL(MISC);
+
+template <SIZE_T MEMSIZE>
+class StackString
+{
+private:
+    char m_innerBuffer[MEMSIZE];
+    char *m_buffer;
+	SIZE_T m_allocation; // actual allocated size
+
+public:
+	StackString(SIZE_T size)
+	: m_allocation(0), m_buffer(NULL)
+	{
+		ReallocateBuffer(size);
+	}
+	
+	StackString()
+	: m_allocation(0), m_buffer(NULL)
+	{
+	}
+
+	void Set(const char *buffer, SIZE_T size)
+	{
+		Resize(size);
+		CopyMemory(m_buffer, buffer, size);
+	}
+
+	void Set(const StackString &s)
+	{
+		Resize(s.m_allocation);
+		CopyMemory(m_buffer, s.m_buffer, s.m_size);
+	}
+	
+	SIZE_T getSize()
+	{
+	    return m_allocation;
+	}
+	
+private:
+    StackString(const StackString &s)
+    {   
+        Set(s);
+    }
+    
+	void Resize(SIZE_T size)
+	{
+		if (size > MEMSIZE && size > m_allocation)
+		{
+			ReallocateBuffer(size);
+		}
+	}
+
+	void ReallocateBuffer(SIZE_T allocation)
+	{
+		if (allocation >= 0 && allocation <= MEMSIZE)
+		{
+		    DeleteBuffer();
+			m_buffer = m_innerBuffer;
+		}
+	    else
+		{
+		    char *newBuffer = (char*)PAL_malloc(allocation);
+		    if (newBuffer == NULL)
+            {
+                ERROR("malloc failed\n");
+                SetLastError(ERROR_NOT_ENOUGH_MEMORY);
+                free(newBuffer);
```

Comment:
free returns immediately on NULL, this is not going to throw, but will remove this if it's going to cause any perf issues.